engine/flink/management/periodic/src/main/scala/pl/touk/nussknacker/engine/management/periodic/PeriodicProcessManager.scala

```diff
@@ -43,14 +43,16 @@ object PeriodicProcessManager {
     val (db: jdbc.JdbcBackend.DatabaseDef, dbProfile: JdbcProfile) = DbInitializer.init(periodicBatchConfig.db)
     val scheduledProcessesRepository = new SlickPeriodicProcessesRepository(db, dbProfile, clock)
     val jarManager = FlinkJarManager(flinkConfig, periodicBatchConfig, modelData, enrichDeploymentWithJarDataFactory(originalConfig))
-    val service = new PeriodicProcessService(delegate, jarManager, scheduledProcessesRepository, listenerFactory.create(originalConfig), additionalDeploymentDataProvider, clock)
+    val listener = listenerFactory.create(originalConfig)
+    val service = new PeriodicProcessService(delegate, jarManager, scheduledProcessesRepository, listener, additionalDeploymentDataProvider, clock)
     system.actorOf(DeploymentActor.props(service, periodicBatchConfig.deployInterval))
     system.actorOf(RescheduleFinishedActor.props(service, periodicBatchConfig.rescheduleCheckInterval))
 
     val toClose = () => {
       Await.ready(system.terminate(), 10 seconds)
       db.close()
       Await.ready(backend.close(), 10 seconds)
+      listener.close()
```

Comment:
maybe it's better to close listener first? And/or wrap in some try/catch?

Reply:
hmm, but as I can read in comments in the code:

> Listener is invoked during DB transaction,

Doesn't it imply that listers should be closed after db? 

As for wraping it in try catch, I will do that

Reply:
I'd say that we should close in reverse order than we create. But - I was more thinking about order WRT akka/backend